users/views.py

```diff
@@ -20,23 +21,41 @@ class LoginView(TemplateView):
     def get(self, request, *args, **kwargs):
         next_url = request.GET.get('next')
         app = None
+        oidc_client = None
+
         if next_url:
             # Determine application from the 'next' query argument.
             # FIXME: There should be a better way to get the app id.
             params = parse_qs(urlparse(next_url).query)
             client_id = params.get('client_id')
+
             if client_id and len(client_id):
                 client_id = client_id[0].strip()
+
             if client_id:
                 try:
                     app = get_application_model().objects.get(client_id=client_id)
                 except get_application_model().DoesNotExist:
                     pass
+
+                try:
+                    oidc_client = Client.objects.get(client_id=client_id)
+                except get_application_model().DoesNotExist:
```

Comment:
This should probably be Client.DoesNotExist

Reply:
Though IMHO `oidc_client = Client.objects.filter(client_id=client_id).first()` would be cleaner anyway.

Reply:
Yeah it should. I'll fix it.